ui/helpers/utils/util.js

```diff
@@ -750,3 +750,30 @@ export const hexToText = (hex) => {
 export const getAvatarFallbackLetter = (subjectName) => {
   return subjectName?.match(/[a-z0-9]/iu)?.[0] ?? '?';
 };
+
+/**
+ * Get Snap permissions filtered by weight.
+ *
+ * @param weightedPermissions - Set of Snap permissions that have 'weight' property assigned.
+ * @param weightThreshold - Number that represents weight threshold for filtering.
+ * @param showFirstThreeOnEmpty - Boolean flag to control if first three permissions should be shown
+ * when no permissions meet the weight criteria.
+ * @returns Subset of permissions passing weight threshold.
+ */
+export const getFilteredSnapPermissions = (
+  weightedPermissions,
+  weightThreshold,
+  showFirstThreeOnEmpty = false,
+) => {
+  let filteredPermissions = weightedPermissions.filter(
+    (permission) => permission.weight <= (weightThreshold ?? Infinity),
+  );
+
+  // If there are no permissions that fall into desired set filtered by weight,
+  // then show only the first three, no matter what the weight is
+  if (showFirstThreeOnEmpty && filteredPermissions.length === 0) {
+    filteredPermissions = weightedPermissions.slice(0, 3);
+  }
```

Comment:
I had discussion about this two times with Erik, he wanted to show only the high risk one at the top level, even it's just a single one, the rest remains hidden under "show all". Here can be observed that example: https://metamask.github.io/snap-simple-keyring/1.1.1/

The logic added here in this function is for the case when we cannot really show anything because all permissions provided are not falling into the desired weight category. So, it's just to prevent empty UI state with "show all" option visible.

Reply:
I'm not sure about this decision, I definitely thought we agreed to only apply filtering once we had a certain amount of permissions.

It would also feel a bit weird to only have 1 permission with a show all button IMO 🤔 

Wondering if we should go with the original plan and revisit when Erik gets back?

Reply:
This is revisited on the meeting we had. So, the decision was to always show at least `3` permissions. Updates are made and changes pushed to this PR.